### DISCUSSION TITLE
Resolution

### DISCUSSION THREAD

**Discussion by joelmce** [koenderks/aRtsy#157] [Q&A] [answered]
Hey there,
I'm testing out the resolution parameters for the flow function, but it seems that no matter what integer I use I am received with the same output in terms of resolution quality. Is this a bug or is there something I'm missing?

Many thanks,
Joel

**Comment by koenderks** [comment, marked as answer]
Hey!

Yeah good point... I believe the purpose of the resolution parameter in this function was to allow for a larger canvas (e.g., the idea was that using more angles should yield more fine grained/flexible lines), but since this algorithm using a (fixed) maximum step size of `resolution * 0.01` the resolution parameter does not have the intended effect. Instead, it might be a better idea to replace this parameter with something like `stepmax`, which allows you to set the maximum step size for each iteration of each line (i.e., larger values of `stepmax` yield longer lines and smaller values yield shorter lines). I'll play around with this. For example, the output may vary like this:

`canvas_flow(colors = colorPalette("dark2"), stepmax = 0.01)`
![stepdot01](https://user-images.githubusercontent.com/25059399/137580592-cc9a8376-c279-4693-a26d-13006a38271a.png)

`canvas_flow(colors = colorPalette("dark2"), stepmax = 0.001)`
![stepdot001](https://user-images.githubusercontent.com/25059399/137580604-bac5ff1b-c545-4a42-afb4-614fe15d31e4.png)
`

By the way, if you want to increase the resolution quality of the output, it is advised to do this when exporting the plot. For example, the `saveCanvas()` function (which is basically a wrapper for `ggsave()`) uses a standard quality of `dpi = 300`, but if you set this to a higher number the quality of the image will increase.

```
x <- canvas_flow(colors = colorPalette("dark2"))
saveCanvas(x, filename = "x.png", dpi = 300) # Standard quality = ggsave(x, filename = "x.png", height = 7, widht = 7, dpi = 300, units = "cm")
saveCanvas(x, filename = "x2.png", dpi = 1000) # Better quality = ggsave(x, filename = "x.png", height = 7, widht = 7, dpi = 1000, units = "cm")
```

**Comment by joelmce** [comment]
I see now, appreciate the rapid response. Love your work by the way!
